Add EnrollmentUrl to Did document

Izzzu opened the pull request:
co-author: algattik

## What this PR changes/adds

Include Registration service enrollment api endpoint in Dataspace Authority DID document.

## Why it does that

The Enrollment API url should be resolved from the Dataspace DID. A subsequent PR will implement that.

## Further notes

Example DID: https://mvd144dataspacedid.z16.web.core.windows.net/.well-known/did.json

Also:
- Corrected some variable names "DID URL" into "DID URI"
- Simplified Terraform outputs by having the "/authority" registration service URL prefix contained in the output registration service URL

## Linked Issue(s)

Linked to https://github.com/agera-edc/MinimumViableDataspace/issues/43

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly?

Izzzu committed:
add enrollment url to did

Izzzu committed:
test - deploy only did

Izzzu committed:
uncomment

Izzzu committed:
uncomment

cpeeyush commented on deployment/terraform/dataspace/main.tf:
```diff
@@ -40,8 +40,10 @@ locals {
+  dataspace_did_uri = "did:web:${azurerm_storage_account.dataspace_did.primary_web_host}"
+  gaiax_did_uri     = "did:web:${azurerm_storage_account.gaiax_did.primary_web_host}"
+
+  registration_service_url = "http://${azurerm_container_group.registration-service.fqdn}:${local.edc_default_port}"
```
Due to recent changes related to JWS now the /authority endpoints in registration service are running on a different port (8182) and not on the edc default ports.

See change in this MVD PR which also adjusting registration URL as per new port https://github.com/agera-edc/MinimumViableDataspace/pull/178/files#diff-b87792b7f7cb2c655edf3945f48b0a96729a8c73f9de2e42d2715d547a4e5a9e.

JWS change on registration service already merged in main branch so it will be a good idea to use port 8182 here as done in above mentioned PR.

cpeeyush replied:
Maybe we can align/add changes as per PR #178 so it will not create many conflicts and will be easier to resolve.

Izzzu replied:
What about waiting for which Pr will get merged first and then resolve the conflicts by applying the changes from #178?
PR #178 is still in review and it may change so it can be difficult to adapt this one with the changes

cpeeyush commented on deployment/terraform/dataspace/main.tf:
```diff
@@ -151,11 +153,18 @@ resource "azurerm_storage_blob" "dataspace_did" {
+    "service" : [
+      {
+        "id" : "#enrollment-url",
+        "type" : "EnrollmentUrl",
```
nit: maybe we can say registration-service-url as this is the base URL for registration service.
```suggestion
        "id" : "#registration-service-url",
        "type" : "RegistrationServiceUrl",
```

Izzzu replied:
what about `RegistrationUrl`?

Izzzu replied:
Changed to RegistrationUrl after discussing offline

cpeeyush commented on deployment/terraform/dataspace/main.tf:
```diff
@@ -151,11 +153,18 @@ resource "azurerm_storage_blob" "dataspace_did" {
+      {
+        "id" : "#enrollment-url",
+        "type" : "EnrollmentUrl",
+        "serviceEndpoint" : "${local.registration_service_url}/api"
```
Because of JWS addition in registration service.
```suggestion
        "serviceEndpoint" : "${local.registration_service_url}/authority"
```

Izzzu committed:
Update deployment/terraform/dataspace/main.tf

Co-authored-by: Peeyush Chandel <555114+cpeeyush@users.noreply.github.com>

Izzzu committed:
changes after review

Izzzu committed:
revert

Izzzu committed:
Changes after review

Izzzu committed:
Use the new reg service cli from branch

Use the new reg service cli from branch

Use the new reg service cli from branch

Update edc

Update reg service version

Izzzu committed:
Revert "Use the new reg service cli from branch"

This reverts commit 322d03cb54b115b0032573a0a59094b5fcf1d523.

algattik committed:
Feature/14/14 Registration Service JWS authentication (#178)

* Use Registration Service CLI to onboard participants (#154)

* .

* .

* .

* .

* .

* .

* .

* .

* .

* PR comments

* Update cd.yaml

* adapted CLI call in VerifyLocalTests

* Update register-participants.sh

* Update register-participants.sh

* Fixed DID URLs for docker-compose

* Adapted for upstream changes

* Update JWT_AUDIENCE to localhost due to docker hosts and registration script differences

* Revert "Update JWT_AUDIENCE to localhost due to docker hosts and registration script differences"

This reverts commit ad831731f37720bfd102ce6cc19640616661795e.

* use docker for cli tools

Use docker for cli tools

add execute permission explicitly

* Simplified scripts

* Update doc to reflect cli-tools changes

* Update version number

* use latest reg service

Co-authored-by: Peeyush Chandel <555114+cpeeyush@users.noreply.github.com>

algattik committed:
Merge branch 'feature/14-rs-jws' into feature/43/43-enrollment-url-in-did

algattik committed:
Simplified use of registration_service_url

algattik committed:
Update main.tf

algattik committed:
Merge branch 'feature/43-enrollment-url-in-did' into feature/43/43-enrollment-url-in-did

algattik committed:
Update cloud-cd.yaml

algattik committed:
Revert "Update cloud-cd.yaml"

This reverts commit c8b0b9e69d76f365c7662710c17b93023e73497c.

cpeeyush commented on deployment/terraform/dataspace/main.tf:
```diff
@@ -86,7 +86,7 @@ resource "azurerm_container_group" "registration-service" {
     environment_variables = {
       EDC_CONNECTOR_NAME      = local.connector_name
-      JWT_AUDIENCE            = "${local.registration_service_url}${local.registration_service_path_prefix}"
+      JWT_AUDIENCE            = local.registration_service_url
```
we also need to modify it for local setup ? https://github.com/agera-edc/MinimumViableDataspace/blob/3ec52c9e798f76b63008ea92ba5799220e2fde07/system-tests/docker-compose.yml#L144

or we plan to do it in subsequent PR ?

algattik replied:
No update needed, the URL still has /authority in it. We just assemble it differently in TF

cpeeyush commented on .github/workflows/deploy.yaml:
```diff
@@ -285,7 +285,7 @@ jobs:
           application_sp_object_id = "${{ secrets.APP_OBJECT_ID }}"
           application_sp_client_id = "${{ secrets.APP_CLIENT_ID }}"
-          registration_service_api_url = "${{ needs.Deploy-Dataspace.outputs.registration_service_url }}/authority"
+          registration_service_api_url = "${{ needs.Deploy-Dataspace.outputs.registration_service_url }}"
```
As we are not upgrading registration service dependency in this PR, so this change is not needed, correct ?

cpeeyush commented on .github/workflows/deploy.yaml:
```diff
@@ -396,7 +396,7 @@ jobs:
             --ids-url "http://${{ env.EDC_HOST }}:8282"
         env:
-          REGISTRATION_SERVICE_API_URL: ${{ needs.Deploy-Dataspace.outputs.registration_service_url }}/authority
+          REGISTRATION_SERVICE_API_URL: ${{ needs.Deploy-Dataspace.outputs.registration_service_url }}
```
As we are not upgrading registration service dependency in this PR, so this change is not needed, correct ?

Izzzu replied:
as I understand this change is only a refactoring because registration_service_url contains now the path prefix, so effectively nothing changes in this line
<img width="1396" alt="Screenshot 2022-07-29 at 09 16 50" src="https://user-images.githubusercontent.com/4217554/181704431-70e5a618-d5c5-423e-a333-b5492054aec8.png">

cpeeyush replied:
Thanks @Izzzu

Izzzu reviewed (commented):
@algattik approving

cpeeyush reviewed (approved):
Approved with suggestion
